### ISSUE TITLE
🐛 Bug Report:  Traceloop does not work in python 3.8

### ISSUE THREAD

**Issue by JoshFerge** [traceloop/openllmetry#197]
### Which component is this bug for?

Traceloop SDK

### 📜 Description

on `3.8.13`, i get the error 

```
    class TracerWrapper(object):
  File "/Users/jferge/code/ai/python-demo-project/venv/lib/python3.8/site-packages/traceloop/sdk/tracing/tracing.py", line 155, in TracerWrapper
    headers: dict[str, str],
TypeError: 'type' object is not subscriptable
```

it appears that this is python 3.9 syntax, and needs `from __future__ import annotations` for 3.8. 

In the project's pyproject.toml, it appears 3.8 should be supported. are multiversion tests being ran?

https://github.com/traceloop/openllmetry/blob/main/packages/traceloop-sdk/pyproject.toml

### 👟 Reproduction steps

use project in python 3.8

### 👍 Expected behavior

should not error upon import

### 👎 Actual Behavior with Screenshots

errors upon import

### 🤖 Python Version

3.8.13

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by galkleinman** [comment]
Thanks for reporting this one @JoshFerge! 💪 

I'm looking into this one, will keep you posted here with the progress.

**Comment by galkleinman** [comment]
@JoshFerge I merged a fix PR (linked to this issue). Python 3.8 should be properly supported right now.

I had also added in the same PR (thanks to your suggestion) a test matrix in the CI for python version >=3.8 <3.12.
Currently 3.12 is out of the matrix as a result of an issue with `numpy` which is a test only dependency in `traceloop-sdk`, which is introducing an issue with our supported version matrix (>=1.25 deprecated the support for python 3.8 and <1.26 doesn't support python 3.12, while we aim to support 3.8-3.12 inclusive). I'll open a related issue for this one.

Thanks again for contributing and raising this issue ❤️
